### ISSUE TITLE
Month and year not focusable using tab in firefox

### ISSUE THREAD

**Issue by akzhy** [palantir/blueprint#5813]
#### Environment

- __Package version(s)__:  ^3.15.1
- __Operating System__: macOS Ventura 13.0.1
- __Browser name and version__: Firefox 108.0.1

#### Code Sandbox

Link to a minimal repro: https://codesandbox.io/s/jolly-nash-tyt67e?file=/src/index.tsx

#### Steps to reproduce

1. Press TAB until month / year is focused

#### Actual behavior

On Firefox, focus never reaches month/year

#### Expected behavior

Month/Year should be focusable by pressing Tab

#### Possible solution

Make it focusable?

#### Bug preview

![](https://i.ibb.co/RB2hvrJ/firefox.gif)

**Comment by adidahiya** [comment]
Confirmed, I can repro this bug with @blueprintjs/datetime v4.4.x
